partiql-eval/src/plan.rs

```diff
@@ -187,6 +187,40 @@ impl EvaluatorPlanner {
                 let to = self.plan_values(*expr.to);
                 Box::new(EvalBetweenExpr { value, from, to })
             }
+            ValueExpr::SimpleCase(e) => {
+                let cases = e
+                    .cases
+                    .into_iter()
+                    .map(|case| {
+                        (
+                            self.plan_values(ValueExpr::BinaryExpr(
+                                BinaryOp::Eq,
+                                e.expr.clone(),
+                                case.0,
+                            )),
+                            self.plan_values(*case.1),
+                        )
+                    })
+                    .collect();
+                let default = e
+                    .default
+                    .as_ref()
+                    .map(|default| self.plan_values(*default.clone()));
+                // Here, rewrite `SimpleCaseExpr`s as `SearchedCaseExpr`s
```

Comment:
Modeling simple and searched case expressions separately in the logical plan but reducing simple case expressions to searched case expressions in the evaluator.